contracts/core/src/contract.rs

```diff
@@ -154,21 +175,11 @@ fn query_exchange_rate(deps: Deps<NeutronQuery>, config: &Config) -> ContractRes
             denom: LD_DENOM.load(deps.storage)?,
         }))?;
 
-    let delegations_response = deps
-        .querier
-        .query_wasm_smart::<drop_staking_base::msg::puppeteer::DelegationsResponse>(
-        &config.puppeteer_contract,
-        &drop_puppeteer_base::msg::QueryMsg::Extension {
-            msg: drop_staking_base::msg::puppeteer::QueryExtMsg::Delegations {},
-        },
-    )?;
+    let exchange_rate_denominator = ld_total_supply.amount.amount;
```

Comment:
This variable is being shadowed on line 211. That's 100% wrong. You shouldn't define denominator this early in this function.